erdos/graph.py

```diff
@@ -189,6 +189,22 @@ def _build_refined_op_graph(self):
             op_handle.op_cls.setup_streams(
                 DataStreams(op_handle.input_streams), **op_handle.setup_args)
 
+        # make stream name unique
```

Comment:
This logic should not happen in `graph.py`. IMHO, each data stream should have a uid, which should be deterministically generated (determinism is useful for debugging). Then we'll have to ensure that we use `stream.get_uid()` instead of `stream.name` in the places we index by stream name. Moreover, we won't have to de-unify streams in `*_output_streams` as you do now. What do you think?